packages/mcp-prompts/prompts/generate-terraform-module-diagram.md

```diff
@@ -0,0 +1,352 @@
+2. **Extract resources** and identify provider types
+3. **Create flowchart structure**:
+   - Use `flowchart LR` for modules with 5+ resources
+   - Organize resources into logical subgraphs/layers
+   - Add appropriate icons for each resource
+4. **Generate Mermaid code** with minimal arrows (connect layers, not individual resources)
+5. **Read current README.md** to find `## Diagram` section or `<!-- BEGIN_TF_DOCS -->` marker
+6. **ACTUALLY MODIFY the README.md file** using `replace_string_in_file` tool:
+   - If `## Diagram` exists: replace the Mermaid code block
+   - If not: insert complete section before `<!-- BEGIN_TF_DOCS -->`
+   - Do NOT just show the code - make the actual file modification
+7. **Verify the tool execution succeeded**
+8. **Do NOT add any legend, notes, or explanatory text after the diagram** - only the Mermaid code block
+
+**You MUST use the file editing tools to make actual changes. Showing the diagram without modifying the file is not sufficient.**
+
+**After completion, provide a brief summary with**:
+
+- List of main resources included in the diagram
+- **Confirmation with file path that README.md was ACTUALLY modified** (not "I would update" or "Here's the code")
+- Any issues or limitations encountered
+
+---
+
+## 🎯 Example Output
+
+For a module with Azure resources, prefer this approach:
+
+```mermaid
+flowchart LR
+    subgraph Network["Network Layer"]
+        vnet["Virtual Network"]
+        vnet@{ icon: "azure:virtual-networks"}
+
+        subnet["Subnet"]
+    end
+
+    subgraph Compute["Compute Layer"]
+        asp["App Service Plan"]
+        asp@{ icon: "azure:app-service-plans"}
+
+        func["Function App"]
+        func@{ icon: "azure:function-apps"}
+    end
+
+    subgraph Storage["Storage Layer"]
+        st["Storage Account"]
+        st@{ icon: "azure:storage-accounts"}
+    end
+
+    subgraph Monitoring["Monitoring"]
+        ai["Application Insights"]
+        ai@{ icon: "azure:application-insights"}
+    end
+
+    %% Show dependencies between layers
+    vnet --> func
+    asp --> func
+    func --> st
+    ai -.-> func
+```
+
+**Why this approach**:
+
+- **LR layout** prevents vertical crowding and crossing arrows
+- **Organized in logical layers** showing the architecture
+- **Focus on resources and their dependencies**, not input/output flow
+- **Minimal arrows** - only essential relationships shown
+- No generic "Module Inputs/Outputs" blocks - the diagram shows the actual infrastructure
+
+**Important Notes**:
+
+- Always verify icon names in the official documentation before generating the final diagram
+- **Do NOT apply fill colors (classDef with fill) to cloud provider icon nodes** - this will hide the icons
+- **Do NOT add "Module Inputs" or "Module Outputs" nodes** - focus on actual infrastructure resources
+- For complex modules with many resources, prefer showing high-level relationships between layers/subgraphs
```

Comment:
thoughts: what about something like:

```
* **Complexity Control**: If >15 resources, simplify by grouping minor resources (e.g., hide NICs, Disks inside the VM logic implicitly, ...) or connecting subgraphs instead of every single node.
```